### PR TITLE
Support Bazel 0.24.1

### PR DESCRIPTION
**Pull request by evie404**
Includes:

- deprecation fixes
- updates dependencies to newer versions like `protobuf`, `rules_node`, and `rules_closure`
- removes C# support. the referenced `rules_dotnet` was too old and fixing it was too much work

Open to cutting it into smaller, more targeted PRs if needed.

Build passes on Travis: https://travis-ci.org/vsco/rules_protobuf/builds/573502159

### CODE REVIEW

File: WORKSPACE [hunk 1]
```diff
@@ -38,48 +39,27 @@ gazelle_dependencies()
 # ================================================================
 
 
-http_archive(
+git_repository(
```
Comment (promiseofcake) [inline WORKSPACE:42]: Should we use the `github_archive` paradigm here as well?

**Review by promiseofcake — APPROVED** [review]
Nicely done, thanks.

**Review by akakitani — APPROVED** [review]

File: WORKSPACE (hunk 1 above)
Comment (evie404) [reply WORKSPACE:42]: it seems like `github_archive` is just light wrapper that doesn't add that much functionality. i'm not anchored one way or another

Comment (evie404) [reply WORKSPACE:42]: can do as follow-up if requested

Comment (promiseofcake) [reply WORKSPACE:42]: Not needed, thanks

**Pull request merged**